### ISSUE TITLE
Enable ccache / sccache on our Github CI Actions

### ISSUE THREAD

**Issue by gpshead** [python/core-workflow#436]
This should significantly reduce build latency, particularly rebuild latency when minor changes are pushed in PR branches.

It can help across builds, but as the main branch point of a given PR branch is not synchronized how much it helps there will be limited. A lot more likely to help on release branches for bugfix backports as C code in those trees doesn't evolve quickly.

https://github.com/hendrikmuhs/ccache-action has instructions.  sccache even claims to support Windows (I've never used it).

**Comment by ammaraskar** [comment]
Looks like we even have the ccache action already hovering around in the workflow from here: https://github.com/python/cpython/pull/29935/files

Just need to use it for the C building bits.

**Comment by ammaraskar** [comment]
Actually, since the workflow uses the approach that adds the `ccache` folder to `PATH`:

https://github.com/python/cpython/blob/788154919c2d843a0a995994bf2aed2d074761ec/.github/workflows/build.yml#L67-L68

```yaml
      - name: Add ccache to PATH
        run: echo "PATH=/usr/lib/ccache:$PATH" >> $GITHUB_ENV
```

we might already be using ccache for all the subsequent steps including `Build CPython`.

**Comment by ammaraskar** [comment]
Yeah, we definitely already are thanks to @tiran. Here's a comparison of the same commit built on python/cpython with a full cache vs my personal fork with no cache:

## python/cpython

![original build time](https://user-images.githubusercontent.com/773529/160640943-0d292f91-faf8-416d-b15f-5330850f5459.png)


## ammaraskar/cpython

![fork build time](https://user-images.githubusercontent.com/773529/160639768-9f327c25-fd9c-4d8e-a4ec-c51f20613941.png)

----

I think the action items here are adding the ccache action to the docs and mac build to speed them up and maybe looking at swapping out for sccache so it can be used on Windows as well.

**Comment by erlend-aasland** [comment]
We've been using ccache in our CI for a while. Should we keep this open for the Windows CI?

**Comment by AA-Turner** [comment]
Can we close this? Is Windows still outstanding?
